### DISCUSSION TITLE
nvidia driver problem

### DISCUSSION THREAD

**Discussion by 3eka** [OpenIndiana/oi-userland#7266] [General]
Hi all,

is it only me using Nvidia drivers? 
I cannot make OI nvidia driver working. At the same time, same driver from NVIDIA site (https://www.nvidia.com/en-us/drivers/unix/solaris-display-archive/) does!

so, I did (based on docs.openindiana.org instructions) replacement (current BE runs NVIDIA site driver):
```
[2021-11-08 11:33:05] :; pkginfo -l NVDAgraphics | grep VERSION
   VERSION:  470.74,REV=2021.09.13.23.13
[2021-11-08 11:33:15] pfexec bash
[2021-11-08 11:33:41] beadm create oi_211108.oi_nvidia
[2021-11-08 11:33:59] beadm mount oi_211108.oi_nvidia /mnt/be
[2021-11-08 11:34:34] pkg -R /mnt/be info -l nvidia
[2021-11-08 11:35:21] pkg -R /mnt/be install -vn nvidia-470
[2021-11-08 11:37:09] pkgrm -R /mnt/be NVDAgraphics NVDAgraphicsr
[2021-11-08 11:37:22] pkg -R /mnt/be install -v nvidia-470
[2021-11-08 11:39:09] beadm umount -f /mnt/be
[2021-11-08 11:39:40] beadm activate oi_211108.oi_nvidia
[2021-11-08 11:39:47] init 6
```
System booted, but w/o graphics (lightdm dies). Some analysis I have tried to perform):
```
:; cat /var/log/lightdm/seat0-greeter.log
** (process:1818): WARNING **: 11:46:38.987: Error getting XDG_RUNTIME_DIR from ConsoleKit: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: Method "GetXDGRuntimeDir" with signature "" on interface "org.freedesktop.ConsoleKit.Session" doesn't exist

** Message: 11:46:39.003: Starting lightdm-gtk-greeter 2.0.7 (Mar 29 2020, 16:40:34)

** (process:1825): WARNING **: 11:46:39.004: [Configuration] Failed to read file '/etc/lightdm/lightdm-gtk-greeter.conf': Key file does not start with a group
ld.so.1: lightdm-gtk-greeter: fatal: libEGL.so.1: open failed: No such file or directory
ld.so.1: lightdm-gtk-greeter: fatal: relocation error: file /usr/lib/64/libcairo.so.2: symbol eglGetProcAddress: referenced symbol not found
ld.so.1: lightdm-gtk-greeter: fatal: relocation error: file /usr/lib/64/libgtk-3.so.0: symbol cairo_pattern_destroy: referenced symbol not found
ld.so.1: lightdm-gtk-greeter: fatal: relocation error: file /usr/sbin/lightdm-gtk-greeter: symbol gtk_init: referenced symbol not found

:; locate libEGL.so.1
/opt/local/lib/libEGL.so.1
/opt/local/lib/libEGL.so.1.0.0
/usr/X11/lib/NVIDIA/libEGL.so.1
/usr/X11/lib/NVIDIA/amd64/libEGL.so.1
/usr/lib/libEGL.so.1
/usr/lib/GL/libEGL.so.1
/usr/lib/GL/amd64/libEGL.so.1
/usr/lib/amd64/libEGL.so.1
/usr/lib/mesa/libEGL.so.1
/usr/lib/mesa/libEGL.so.1.0.0
/usr/lib/mesa/amd64/libEGL.so.1
/usr/lib/mesa/amd64/libEGL.so.1.0.0

:; ls -al /usr/lib/libEGL.so.1
lrwxrwxrwx 1 root root 29 Sep 29 15:15 /usr/lib/libEGL.so.1 -> ../X11/lib/NVIDIA/libEGL.so.1

:; pkg search -l /usr/lib/libEGL.so.1
INDEX      ACTION VALUE               PACKAGE
path       link   usr/lib/libEGL.so.1 pkg:/service/opengl/ogl-select@0.5.11-2020.0.1.5

:; pkg search -l /usr/X11/lib/NVIDIA/amd64/libEGL.so.1 
INDEX      ACTION VALUE                                PACKAGE
path       file   usr/X11/lib/NVIDIA/amd64/libEGL.so.1 pkg:/driver/graphics/nvidia-470@0.470.74-2020.0.1.0

:; pkg search -l /usr/lib/GL/amd64/libEGL.so.1
INDEX      ACTION VALUE                        PACKAGE
path       link   usr/lib/GL/amd64/libEGL.so.1 pkg:/service/opengl/ogl-select@0.5.11-2020.0.1.5

:; pkg search -l /usr/lib/mesa/amd64/libEGL.so.1.0.0
INDEX      ACTION VALUE                              PACKAGE
path       file   usr/lib/mesa/amd64/libEGL.so.1.0.0 pkg:/x11/library/mesa@13.0.6-2020.0.1.7
```

Anyone has an idea what to do?

**Comment by AndWac** [comment]
Hi,

I have NVIDIA graphics on all of my OI machines: Quadro K620, P600, and
P620. The K620 is supported by our default NVIDIA driver version 390,
the others need a newer ones.
I don't have many local installed files in /opt, though. I try to create
official packages for all software I am interested in. This helps to
detect errors early.
As my installations are all fine with different versions of NVIDIA
drivers I suspect your additional installations may be the source of the
problems you encounter.

Regards,
Andreas

Am 11/8/21 um 12:13 PM schrieb 3eka:

**Reply by 3eka** [reply]
Packages found (installed) into `/opt/local` are ones from Joyent (https://pkgsrc.joyent.com/install-on-illumos/).
That path is not present in either variable (just hanging there, and using some missing in OI).

**Comment by AndWac** [comment]
Am 11/8/21 um 12:13 PM schrieb 3eka:
>
> Hi all,
>
> is it only me using Nvidia drivers?
> I cannot make OI nvidia driver working. At the same time, same driver
> from NVIDIA site
> (https://www.nvidia.com/en-us/drivers/unix/solaris-display-archive/
> <https://www.nvidia.com/en-us/drivers/unix/solaris-display-archive/>)
> does!
>
> so, I did (based on docs.openindiana.org instructions) replacement
> (current BE runs NVIDIA site driver):
>
> |[2021-11-08 11:33:05] :; pkginfo -l NVDAgraphics | grep VERSION
> VERSION: 470.74,REV=2021.09.13.23.13 [2021-11-08 11:33:15] pfexec bash
> [2021-11-08 11:33:41] beadm create oi_211108.oi_nvidia [2021-11-08
> 11:33:59] beadm mount oi_211108.oi_nvidia /mnt/be [2021-11-08
> 11:34:34] pkg -R /mnt/be info -l nvidia [2021-11-08 11:35:21] pkg -R
> /mnt/be install -vn nvidia-470 [2021-11-08 11:37:09] pkgrm -R /mnt/be
> NVDAgraphics NVDAgraphicsr [2021-11-08 11:37:22] pkg -R /mnt/be
> install -v nvidia-470 [2021-11-08 11:39:09] beadm umount -f /mnt/be
> [2021-11-08 11:39:40] beadm activate oi_211108.oi_nvidia [2021-11-08
> 11:39:47] init 6 |
>
> System booted, but w/o graphics (lightdm dies). Some analysis I have
> tried to perform):
>
> |:; cat /var/log/lightdm/seat0-greeter.log ** (process:1818): WARNING
> **: 11:46:38.987: Error getting XDG_RUNTIME_DIR from ConsoleKit:
> GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: Method
> "GetXDGRuntimeDir" with signature "" on interface
> "org.freedesktop.ConsoleKit.Session" doesn't exist ** Message:
> 11:46:39.003: Starting lightdm-gtk-greeter 2.0.7 (Mar 29 2020,
> 16:40:34) ** (process:1825): WARNING **: 11:46:39.004: [Configuration]
> Failed to read file '/etc/lightdm/lightdm-gtk-greeter.conf': Key file
> does not start with a group ld.so.1: lightdm-gtk-greeter: fatal:
> libEGL.so.1: open failed: No such file or directory ld.so.1:
> lightdm-gtk-greeter: fatal: relocation error: file
> /usr/lib/64/libcairo.so.2: symbol eglGetProcAddress: referenced symbol
> not found ld.so.1: lightdm-gtk-greeter: fatal: relocation error: file
> /usr/lib/64/libgtk-3.so.0: symbol cairo_pattern_destroy: referenced
> symbol not found ld.so.1: lightdm-gtk-greeter: fatal: relocation
> error: file /usr/sbin/lightdm-gtk-greeter: symbol gtk_init: referenced
> symbol not found :; locate libEGL.so.1 /opt/local/lib/libEGL.so.1
> /opt/local/lib/libEGL.so.1.0.0 /usr/X11/lib/NVIDIA/libEGL.so.1
> /usr/X11/lib/NVIDIA/amd64/libEGL.so.1 /usr/lib/libEGL.so.1
> /usr/lib/GL/libEGL.so.1 /usr/lib/GL/amd64/libEGL.so.1
> /usr/lib/amd64/libEGL.so.1 /usr/lib/mesa/libEGL.so.1
> /usr/lib/mesa/libEGL.so.1.0.0 /usr/lib/mesa/amd64/libEGL.so.1
> /usr/lib/mesa/amd64/libEGL.so.1.0.0 :; ls -al /usr/lib/libEGL.so.1
> lrwxrwxrwx 1 root root 29 Sep 29 15:15 /usr/lib/libEGL.so.1 ->
> ../X11/lib/NVIDIA/libEGL.so.1 :; pkg search -l /usr/lib/libEGL.so.1
> INDEX ACTION VALUE PACKAGE path link usr/lib/libEGL.so.1
> ***@***.*** :; pkg search -l
> /usr/X11/lib/NVIDIA/amd64/libEGL.so.1 INDEX ACTION VALUE PACKAGE path
> file usr/X11/lib/NVIDIA/amd64/libEGL.so.1
> ***@***.*** :; pkg search -l
> /usr/lib/GL/amd64/libEGL.so.1 INDEX ACTION VALUE PACKAGE path link
> usr/lib/GL/amd64/libEGL.so.1
> ***@***.*** :; pkg search -l
> /usr/lib/mesa/amd64/libEGL.so.1.0.0 INDEX ACTION VALUE PACKAGE path
> file usr/lib/mesa/amd64/libEGL.so.1.0.0
> ***@***.*** |
>
> Anyone has an idea what to do?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenIndiana/oi-userland/discussions/7266>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADET4N6GICG2V7NOUFQHXYLUK6WF5ANCNFSM5HSMMM6A>.
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>
This is what my lightdm-gtk-greeter finds. What does yours try to use?

╰─➤  ldd /usr/sbin/lightdm-gtk-greeter

     libgtk-3.so.0 =>     /usr/lib/64/libgtk-3.so.0
     libgdk-3.so.0 =>     /usr/lib/64/libgdk-3.so.0
     libcairo.so.2 =>     /usr/lib/64/libcairo.so.2
     libgdk_pixbuf-2.0.so.0 =>  /usr/lib/64/libgdk_pixbuf-2.0.so.0
     libgio-2.0.so.0 =>     /usr/lib/64/libgio-2.0.so.0
     libgobject-2.0.so.0 =>     /usr/lib/64/libgobject-2.0.so.0
     libglib-2.0.so.0 =>     /usr/lib/64/libglib-2.0.so.0
     liblightdm-gobject-1.so.0 =>  /usr/lib/64/liblightdm-gobject-1.so.0
     libX11.so.4 =>     /usr/lib/64/libX11.so.4
     libm.so.2 =>     /lib/64/libm.so.2
     libc.so.1 =>     /lib/64/libc.so.1
     libXinerama.so.1 =>     /usr/lib/64/libXinerama.so.1
     libXrandr.so.2 =>     /usr/lib/64/libXrandr.so.2
     libXcursor.so.1 =>     /usr/lib/64/libXcursor.so.1
     libXext.so.0 =>     /usr/lib/64/libXext.so.0
     librt.so.1 =>     /lib/64/librt.so.1
     libgmodule-2.0.so.0 =>     /usr/lib/64/libgmodule-2.0.so.0
     libpangocairo-1.0.so.0 =>  /usr/lib/64/libpangocairo-1.0.so.0
     libXi.so.5 =>     /usr/lib/64/libXi.so.5
     libXcomposite.so.1 =>     /usr/lib/64/libXcomposite.so.1
     libXdamage.so.1 =>     /usr/lib/64/libXdamage.so.1
     libXfixes.so.1 =>     /usr/lib/64/libXfixes.so.1
     libcairo-gobject.so.2 =>  /usr/lib/64/libcairo-gobject.so.2
     libatk-1.0.so.0 =>     /usr/lib/64/libatk-1.0.so.0
     libatk-bridge-2.0.so.0 =>  /usr/lib/64/libatk-bridge-2.0.so.0
     libepoxy.so.0 =>     /usr/lib/64/libepoxy.so.0
     libfribidi.so.0 =>     /usr/lib/64/libfribidi.so.0
     libpangoft2-1.0.so.0 =>     /usr/lib/64/libpangoft2-1.0.so.0
     libpango-1.0.so.0 =>     /usr/lib/64/libpango-1.0.so.0
     libharfbuzz.so.0 =>     /usr/lib/64/libharfbuzz.so.0
     libfontconfig.so.1 =>     /usr/lib/64/libfontconfig.so.1
     libfreetype.so.6 =>     /usr/lib/64/libfreetype.so.6
     libpixman-1.so.0 =>     /usr/lib/64/libpixman-1.so.0
     libEGL.so.1 =>     /usr/lib/64/libEGL.so.1
     libdl.so.1 =>     /lib/64/libdl.so.1
     libpng16.so.16 =>     /usr/lib/64/libpng16.so.16
     libX11-xcb.so.1 =>     /usr/lib/64/libX11-xcb.so.1
     libxcb.so.1 =>     /usr/lib/64/libxcb.so.1
     libxcb-render.so.0 =>     /usr/lib/64/libxcb-render.so.0
     libz.so.1 =>     /usr/lib/64/libz.so.1
     libGL.so.1 =>     /usr/lib/64/libGL.so.1
     libsocket.so.1 =>     /lib/64/libsocket.so.1
     libnsl.so.1 =>     /lib/64/libnsl.so.1
     libresolv.so.2 =>     /lib/64/libresolv.so.2
     libffi.so.6 =>     /usr/lib/64/libffi.so.6
     libpcre.so.0 =>     /usr/lib/64/libpcre.so.0
     libxklavier.so.16 =>     /usr/lib/64/libxklavier.so.16
     libXrender.so.1 =>     /usr/lib/64/libXrender.so.1
     libatspi.so.0 =>     /usr/lib/64/libatspi.so.0
     libdbus-1.so.3 =>     /usr/lib/64/libdbus-1.so.3
     libthai.so.0 =>     /usr/lib/64/libthai.so.0
     libgraphite2.so.3 =>     /usr/lib/64/libgraphite2.so.3
     libexpat.so.1 =>     /usr/lib/64/libexpat.so.1
     libuuid.so.1 =>     /lib/64/libuuid.so.1
     libbz2.so.1 =>     /usr/lib/64/libbz2.so.1
     libbrotlidec.so.1 =>     /usr/lib/64/libbrotlidec.so.1
     libpthread.so.1 =>     /lib/64/libpthread.so.1
     libnvidia-glsi.so.1 =>     /usr/lib/64/libnvidia-glsi.so.1
     libXau.so.6 =>     /usr/lib/64/libXau.so.6
     libXdmcp.so.6 =>     /usr/lib/64/libXdmcp.so.6
     libnvidia-tls.so.1 =>     /usr/lib/64/libnvidia-tls.so.1
     libnvidia-glcore.so.1 =>  /usr/lib/64/libnvidia-glcore.so.1
     libmp.so.2 =>     /lib/64/libmp.so.2
     libmd.so.1 =>     /lib/64/libmd.so.1
     libxml2.so.2 =>     /usr/lib/64/libxml2.so.2
     libxkbfile.so.5 =>     /usr/lib/64/libxkbfile.so.5
     libbsm.so.1 =>     /lib/64/libbsm.so.1
     libdatrie.so.1 =>     /usr/lib/amd64/libdatrie.so.1
     libdlpi.so.1 =>     /lib/64/libdlpi.so.1
     libbrotlicommon.so.1 =>     /usr/lib/64/libbrotlicommon.so.1
     liblzma.so.5 =>     /usr/lib/64/liblzma.so.5
     libsecdb.so.1 =>     /lib/64/libsecdb.so.1
     libtsol.so.2 =>     /lib/64/libtsol.so.2
     libinetutil.so.1 =>     /lib/64/libinetutil.so.1
     libscf.so.1 =>     /lib/64/libscf.so.1
     libdladm.so.1 =>     /lib/64/libdladm.so.1
     libuutil.so.1 =>     /lib/64/libuutil.so.1
     libgen.so.1 =>     /lib/64/libgen.so.1
     libnvpair.so.1 =>     /lib/64/libnvpair.so.1
     libsmbios.so.1 =>     /usr/lib/64/libsmbios.so.1
     libdevinfo.so.1 =>     /lib/64/libdevinfo.so.1
     librcm.so.1 =>     /lib/64/librcm.so.1
     libexacct.so.1 =>     /usr/lib/64/libexacct.so.1
     libkstat.so.1 =>     /lib/64/libkstat.so.1
     libpool.so.1 =>     /usr/lib/64/libpool.so.1
     libsec.so.1 =>     /lib/64/libsec.so.1
     libavl.so.1 =>     /lib/64/libavl.so.1
     libidmap.so.1 =>     /usr/lib/64/libidmap.so.1
     libXevie.so.1 =>     /usr/lib/64/libXevie.so.1
     libXss.so.1 =>     /usr/lib/64/libXss.so.1
     libofmt.so.1 =>     /lib/64/libofmt.so.1

**Reply by 3eka** [reply]
Here you are:
```
:; ldd /usr/sbin/lightdm-gtk-greeter 
        libgtk-3.so.0 =>         /usr/lib/64/libgtk-3.so.0
        libgdk-3.so.0 =>         /usr/lib/64/libgdk-3.so.0
        libcairo.so.2 =>         /usr/lib/64/libcairo.so.2
        libgdk_pixbuf-2.0.so.0 =>        /usr/lib/64/libgdk_pixbuf-2.0.so.0
        libgio-2.0.so.0 =>       /usr/lib/64/libgio-2.0.so.0
        libgobject-2.0.so.0 =>   /usr/lib/64/libgobject-2.0.so.0
        libglib-2.0.so.0 =>      /usr/lib/64/libglib-2.0.so.0
        liblightdm-gobject-1.so.0 =>     /usr/lib/64/liblightdm-gobject-1.so.0
        libX11.so.4 =>   /usr/lib/64/libX11.so.4
        libm.so.2 =>     /lib/64/libm.so.2
        libc.so.1 =>     /lib/64/libc.so.1
        libXinerama.so.1 =>      /usr/lib/64/libXinerama.so.1
        libXrandr.so.2 =>        /usr/lib/64/libXrandr.so.2
        libXcursor.so.1 =>       /usr/lib/64/libXcursor.so.1
        libXext.so.0 =>  /usr/lib/64/libXext.so.0
        librt.so.1 =>    /lib/64/librt.so.1
        libgmodule-2.0.so.0 =>   /usr/lib/64/libgmodule-2.0.so.0
        libpangocairo-1.0.so.0 =>        /usr/lib/64/libpangocairo-1.0.so.0
        libXi.so.5 =>    /usr/lib/64/libXi.so.5
        libXcomposite.so.1 =>    /usr/lib/64/libXcomposite.so.1
        libXdamage.so.1 =>       /usr/lib/64/libXdamage.so.1
        libXfixes.so.1 =>        /usr/lib/64/libXfixes.so.1
        libcairo-gobject.so.2 =>         /usr/lib/64/libcairo-gobject.so.2
        libatk-1.0.so.0 =>       /usr/lib/64/libatk-1.0.so.0
        libatk-bridge-2.0.so.0 =>        /usr/lib/64/libatk-bridge-2.0.so.0
        libepoxy.so.0 =>         /usr/lib/64/libepoxy.so.0
        libfribidi.so.0 =>       /usr/lib/64/libfribidi.so.0
        libpangoft2-1.0.so.0 =>  /usr/lib/64/libpangoft2-1.0.so.0
        libpango-1.0.so.0 =>     /usr/lib/64/libpango-1.0.so.0
        libharfbuzz.so.0 =>      /usr/lib/64/libharfbuzz.so.0
        libfontconfig.so.1 =>    /usr/lib/64/libfontconfig.so.1
        libfreetype.so.6 =>      /usr/lib/64/libfreetype.so.6
        libpixman-1.so.0 =>      /usr/lib/64/libpixman-1.so.0
        libEGL.so.1 =>   /usr/lib/64/libEGL.so.1
        libdl.so.1 =>    /lib/64/libdl.so.1
        libpng16.so.16 =>        /usr/lib/64/libpng16.so.16
        libX11-xcb.so.1 =>       /usr/lib/64/libX11-xcb.so.1
        libxcb.so.1 =>   /usr/lib/64/libxcb.so.1
        libxcb-render.so.0 =>    /usr/lib/64/libxcb-render.so.0
        libz.so.1 =>     /usr/lib/64/libz.so.1
        libGL.so.1 =>    /usr/lib/64/libGL.so.1
        libsocket.so.1 =>        /lib/64/libsocket.so.1
        libnsl.so.1 =>   /lib/64/libnsl.so.1
        libresolv.so.2 =>        /lib/64/libresolv.so.2
        libffi.so.6 =>   /usr/lib/64/libffi.so.6
        libpcre.so.0 =>  /usr/lib/64/libpcre.so.0
        libxklavier.so.16 =>     /usr/lib/64/libxklavier.so.16
        libXrender.so.1 =>       /usr/lib/64/libXrender.so.1
        libatspi.so.0 =>         /usr/lib/64/libatspi.so.0
        libdbus-1.so.3 =>        /usr/lib/64/libdbus-1.so.3
        libthai.so.0 =>  /usr/lib/64/libthai.so.0
        libgraphite2.so.3 =>     /usr/lib/64/libgraphite2.so.3
        libexpat.so.1 =>         /usr/lib/64/libexpat.so.1
        libuuid.so.1 =>  /lib/64/libuuid.so.1
        libbz2.so.1 =>   /usr/lib/64/libbz2.so.1
        libbrotlidec.so.1 =>     /usr/lib/64/libbrotlidec.so.1
        libpthread.so.1 =>       /lib/64/libpthread.so.1
        libnvidia-glsi.so.1 =>   /usr/lib/64/libnvidia-glsi.so.1
        libXau.so.6 =>   /usr/lib/64/libXau.so.6
        libXdmcp.so.6 =>         /usr/lib/64/libXdmcp.so.6
        libnvidia-tls.so.1 =>    /usr/lib/64/libnvidia-tls.so.1
        libnvidia-glcore.so.1 =>         /usr/lib/64/libnvidia-glcore.so.1
        libmp.so.2 =>    /lib/64/libmp.so.2
        libmd.so.1 =>    /lib/64/libmd.so.1
        libxml2.so.2 =>  /usr/lib/64/libxml2.so.2
        libxkbfile.so.5 =>       /usr/lib/64/libxkbfile.so.5
        libbsm.so.1 =>   /lib/64/libbsm.so.1
        libdatrie.so.1 =>        /usr/lib/amd64/libdatrie.so.1
        libdlpi.so.1 =>  /lib/64/libdlpi.so.1
        libbrotlicommon.so.1 =>  /usr/lib/64/libbrotlicommon.so.1
        liblzma.so.5 =>  /usr/lib/64/liblzma.so.5
        libsecdb.so.1 =>         /lib/64/libsecdb.so.1
        libtsol.so.2 =>  /lib/64/libtsol.so.2
        libinetutil.so.1 =>      /lib/64/libinetutil.so.1
        libscf.so.1 =>   /lib/64/libscf.so.1
        libdladm.so.1 =>         /lib/64/libdladm.so.1
        libuutil.so.1 =>         /lib/64/libuutil.so.1
        libgen.so.1 =>   /lib/64/libgen.so.1
        libnvpair.so.1 =>        /lib/64/libnvpair.so.1
        libsmbios.so.1 =>        /usr/lib/64/libsmbios.so.1
        libdevinfo.so.1 =>       /lib/64/libdevinfo.so.1
        librcm.so.1 =>   /lib/64/librcm.so.1
        libexacct.so.1 =>        /usr/lib/64/libexacct.so.1
        libkstat.so.1 =>         /lib/64/libkstat.so.1
        libpool.so.1 =>  /usr/lib/64/libpool.so.1
        libsec.so.1 =>   /lib/64/libsec.so.1
        libavl.so.1 =>   /lib/64/libavl.so.1
        libidmap.so.1 =>         /usr/lib/64/libidmap.so.1
        libXevie.so.1 =>         /usr/lib/64/libXevie.so.1
        libXss.so.1 =>   /usr/lib/64/libXss.so.1
        libofmt.so.1 =>  /lib/64/libofmt.so.1
```
Thanks and regards.

**Reply by 3eka** [reply]
Looks like file
```
/usr/lib/64/libEGL.so.1
```
is part of NVIDIA driver (for their site):
```
:; ls -al /usr/lib/64/libEGL.so.1
lrwxrwxrwx 1 root root 38 Sep 29 15:15 /usr/lib/64/libEGL.so.1 -> ../../X11/lib/NVIDIA/amd64/libEGL.so.1

:; grep -r amd64/libEGL.so.1 /var/sadm/pkg/NVDAgraphics*
/var/sadm/pkg/NVDAgraphics/save/pspool/NVDAgraphics/pkgmap:1 f none X11/lib/NVIDIA/amd64/libEGL.so.1 0644 root bin 1244584 53101 1631574029
/var/sadm/pkg/NVDAgraphics/save/pspool/NVDAgraphics/pkgmap:1 s none lib/amd64/libEGL.so.1=../../X11/lib/NVIDIA/amd64/libEGL.so.1
```
Not sure about that file when using OI nvidia driver (rolled back, so I can use GUI).

**Comment by 3eka** [comment]
UPDATE (thanks to @AndWac)
```
:; pfexec pkg verify ogl-select
PACKAGE STATUS
pkg://openindiana.org/service/opengl/ogl-select  ERROR
        link: usr/lib/libEGL.so.1
                ERROR: Target: '../X11/lib/NVIDIA/libEGL.so.1' should be 'GL/libEGL.so.1'
        link: usr/lib/amd64/libEGL.so.1
                ERROR: Target: '../../X11/lib/NVIDIA/amd64/libEGL.so.1' should be '../GL/amd64/libEGL.so.1'
```
That is probably change made by NVIDIA driver. So, proper solution was:
```
:; pfexec bash
:; NEWBE=oi_$(date '+%y%m%d').oi_nvidia

:; beadm create ${NEWBE}
Created successfully

:; beadm mount ${NEWBE} /mnt/be
Mounted successfully on: '/mnt/be'

:; pkg -R /mnt/be info -l nvidia
pkg: info: no packages matching the following patterns you specified are
installed on the system.  Try querying remotely instead:

        nvidia

:; pkgrm -R /mnt/be NVDAgraphicsr NVDAgraphics
The following package is currently installed:
   NVDAgraphicsr  NVIDIA Graphics System Device Driver
                  (i386) 470.74,REV=2021.09.13.23.13

Do you want to remove this package? [y,n,?,q] y

## Removing installed package instance <NVDAgraphicsr>
...
Removal of <NVDAgraphics> was successful.

:; pkg -R /mnt/be install -v nvidia-470
           Packages to install:         1
            Services to change:         2
     Estimated space available:  38.94 GB
Estimated space to be consumed: 593.12 MB
          Rebuild boot archive:       Yes

Changed packages:
openindiana.org
  driver/graphics/nvidia-470
    None -> 0.470.82-2020.0.1.0

Services:
  restart_fmri:
    svc:/application/desktop-cache/desktop-mime-cache:default
    svc:/application/desktop-cache/icon-cache:default

Building new search index                  2016/2016
DOWNLOAD                                PKGS         FILES    XFER (MB)   SPEED
Completed                                1/1         91/91    69.9/69.9  5.1M/s

PHASE                                          ITEMS
Installing new actions                       174/174
Updating package state database                 Done
Updating package cache                           0/0
Updating image state                            Done
Creating fast lookup database                   Done
Reading search index                            Done
Updating search index                            1/1
Updating package cache                           2/2

:; pkg -R /mnt/be verify ogl-select
PACKAGE                                                                 STATUS
pkg://openindiana.org/service/opengl/ogl-select                          ERROR
        link: usr/lib/libEGL.so
                ERROR: Missing: symbolic link does not exist
        link: usr/lib/libEGL.so.1
                ERROR: Missing: symbolic link does not exist
        link: usr/lib/amd64/libEGL.so
                ERROR: Missing: symbolic link does not exist
        link: usr/lib/amd64/libEGL.so.1
                ERROR: Missing: symbolic link does not exist
                
:; pkg -R /mnt/be fix ogl-select
Packages to fix: 1

Repairing: pkg://openindiana.org/service/opengl/ogl-select@0.5.11,5.11-2020.0.1.5:20200330T163649Z
PACKAGE                                                                 STATUS
pkg://openindiana.org/service/opengl/ogl-select                          ERROR
        link: usr/lib/amd64/libEGL.so.1
                ERROR: Missing: symbolic link does not exist
        link: usr/lib/libEGL.so
                ERROR: Missing: symbolic link does not exist
        link: usr/lib/amd64/libEGL.so
                ERROR: Missing: symbolic link does not exist
        link: usr/lib/libEGL.so.1
                ERROR: Missing: symbolic link does not exist
PHASE                                          ITEMS
Updating modified actions                        4/4
Updating package state database                 Done
Updating package cache                           0/0
Updating image state                            Done
Creating fast lookup database                   Done
Updating package cache                           2/2

:; beadm umount -f /mnt/be
Unmounted successfully

:; pfexec beadm activate ${NEWBE}
Activated successfully

:; init 6
```

**Comment by 3eka** [comment]
Maybe to update http://docs.openindiana.org/dev/graphics-stack/#nvidia ?

**Comment by 3eka** [comment]
An issue is still there:
```
:; pfexec less  /var/log/lightdm/seat0-greeter.log

** (process:1815): WARNING **: 15:55:40.082: Error getting XDG_RUNTIME_DIR from ConsoleKit: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: Method "GetXDGRuntimeDir" with signature "" on interface "org.freedesktop.ConsoleKit.Session" doesn't exist

** Message: 15:55:40.098: Starting lightdm-gtk-greeter 2.0.7 (Mar 29 2020, 16:40:34)

** (process:1823): WARNING **: 15:55:40.099: [Configuration] Failed to read file '/etc/lightdm/lightdm-gtk-greeter.conf': Key file does not start with a group
Error:            Meta_L added to symbol map for multiple modifiers
                  Using Mod4, ignoring Mod1.
Error:            Meta_R added to symbol map for multiple modifiers
                  Using Mod4, ignoring Mod1.

(lightdm-gtk-greeter:1823): Gtk-WARNING **: 15:55:41.313: Drawing a gadget with negative dimensions. Did you forget to allocate a size? (node menubar owner GreeterMenuBar)
```
and
```
:; grep -v ^# /etc/lightdm/lightdm-gtk-greeter.conf
indicators=~a11y;~spacer;~host;~session;~clock
[greeter]
```
No idea what is missing there.
